src/test/librados_test_stub/TestMemIoCtxImpl.cc

```diff
@@ -373,19 +372,29 @@ int TestMemIoCtxImpl::sparse_read(const std::string& oid, uint64_t off,
     }
   }
 
+  //interval_set<uint64_t>::const_iterator cit = file->snap_overlap.begin();
+  for(auto cit = file->snap_overlap.begin(); cit != file->snap_overlap.end(); cit++){
+      std::cerr << "  off: " << cit.get_start() << " len: " << cit.get_len() << std::endl;
+  }
+
   RWLock::RLocker l(file->lock);
   len = clip_io(off, len, file->data.length());
+
   if (m != NULL) {
     m->clear();
     if (len > 0) {
       (*m)[off] = len;
```

Comment:
For the unit tests, it's probably safe enough to just populate the entire requested extent even if it doesn't exist. This test stub doesn't track holes, so you wouldn't get a true implementation.